Plugins/Resources/Resources.cpp

```diff
@@ -0,0 +1,282 @@
+#include "Resources.hpp"
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+
+static Resources::Resources* g_plugin;
+
+const int RESMAN_PRIORITY_USER_HAK = 31000000;
+const int RESMAN_PRIORITY_NWSYNC = 40000000;
+
+const uint32_t LOADMODULE_ERROR_COULDNOTLOADHAK = 2;
+
+const int NWSYNC_STATE_DISCOVERY = 0;   // Currently trying to figure out if we need to do something.
+
+const int NWSYNC_STATE_DOWNLOAD_MANIFEST_META_REQUEST = 5;   // We sent the manifest meta request.
+const int NWSYNC_STATE_DOWNLOAD_MANIFEST_META_PROGRESS = 6;   // Waiting for response/downloading.
+const int NWSYNC_STATE_DOWNLOAD_MANIFEST_META_PARSE = 7;   // Parsing and configuring metadata.
+
+const int NWSYNC_STATE_DOWNLOAD_MANIFEST_REQUEST = 10;  // We sent the manifest request.
+const int NWSYNC_STATE_DOWNLOAD_MANIFEST_PROGRESS = 11;  // Downloading the manifest.
+const int NWSYNC_STATE_DOWNLOAD_MANIFEST_PARSE = 12;  // Parsing the manifest in-memory.
+
+const int NWSYNC_STATE_WAIT_PERMISSION_TO_SYNC = 20;  // We're waiting for you/the user to give the go-ahead. Call SetSyncPermission(true) to advance.
+
+const int NWSYNC_STATE_GOT_PERMISSION_TO_SYNC = 21;  // Internal state.
+
+const int NWSYNC_STATE_SYNC_IN_PROGRESS = 100; // We're currently syncing.
+const int NWSYNC_STATE_SYNC_IN_PROGRESS_FLUSH = 101; // We're currently flushing to disk.
+
+const int NWSYNC_STATE_HOUSEKEEPING_IN_PROGRESS = 200; // Do some post-transfer cleanup.
+
+const int NWSYNC_STATE_DONE = 800; // Done! Go ahead.
+const int NWSYNC_STATE_FAILED = 801; // We aborted. Error will be in currentStateMessage.
+
+static Hooks::Hook s_loadModuleStartHook;
+static Hooks::Hook s_readFieldCExoStringHook;
+static Hooks::Hook s_GetListHook;
+
+NWNX_PLUGIN_ENTRY Plugin* PluginLoad(Services::ProxyServiceList* services)
+{
+    g_plugin = new Resources::Resources(services);
+    return g_plugin;
+}
+
+namespace Resources {
+
+Resources::Resources(Services::ProxyServiceList* services)
+: Plugin(services)
+{
+    s_loadModuleStartHook = Hooks::HookFunction(&CNWSModule::LoadModuleStart, &OnLoadModuleStart, Hooks::Order::Late);
+}
+
+uint32_t Resources::OnLoadModuleStart(CNWSModule *pModule, CExoString param_1, int param_2, int param_3, const NWSync::Advertisement &param_4)
+{
+    bool reloadRequired = false;
+    uint32_t retVal = 0;
+
+    if (Config::Get<bool>("USE_CUSTOM_HAK_LIST"))
```

Comment:
This will return a `std::optional<bool>`, which in an `if()` will evaluate to true if a value is set, regardless of which value. Since the docs say this should default to false, you should add a `, false` argument.